elliptic-curve/src/hash2field/expand_msg/xmd.rs

```diff
@@ -53,24 +54,24 @@ where
         let ell = u8::try_from((len_in_bytes + b_in_bytes - 1) / b_in_bytes).map_err(|_| Error)?;
 
         let domain = Domain::xmd::<HashT>(dst);
-        let mut b_0 = HashT::new().chain(GenericArray::<u8, HashT::BlockSize>::default());
+        let mut b_0 = HashT::new().chain_update(GenericArray::<u8, HashT::BlockSize>::default());
 
         for msg in msgs {
-            b_0 = b_0.chain(msg);
+            b_0 = b_0.chain_update(msg);
         }
 
         let b_0 = b_0
-            .chain(len_in_bytes_u16.to_be_bytes())
-            .chain([0])
-            .chain(domain.data())
-            .chain([domain.len()])
+            .chain_update(len_in_bytes_u16.to_be_bytes())
+            .chain_update([0])
+            .chain_update(domain.data())
+            .chain_update([domain.len()])
```

Comment:
I am not sure compiler is able to properly optimize such long chains. To be safe it could be worth to use `HashT::new` and  `update`.

Reply:
Interesting, is it okay then if I just replace all uses of `chain_update`?

Reply:
Went ahead and just removed all `chain_update`s.

Reply:
Yes, `chain_update` is a convenience method defined in terms of `update`.